hack/generator/pkg/codegen/storage/type_converter.go

```diff
@@ -40,11 +46,63 @@ func NewTypeConverter(types astmodel.Types) *TypeConverter {
 	return result
 }
 
-// ConvertDefinition applies our type conversion to a specific type definition
-func (t *TypeConverter) ConvertDefinition(def astmodel.TypeDefinition) (astmodel.TypeDefinition, error) {
-	result, err := t.visitor.VisitDefinition(def, nil)
+// ConvertResourceDefinition applies our type conversion to a specific resource type definition
+func (t *TypeConverter) ConvertResourceDefinition(def astmodel.TypeDefinition) (astmodel.TypeDefinition, error) {
```

Comment:
It's not clear to me how this function and `ConvertObjectDefinition` are different? I don't see any resource specific handling here?

Reply:
They were going to become different, in another WIP branch. But now I've switched to separate stages, they are the same and I've collapsed them into a single method.